# Numbers in task names  
The uses and meanings of numbers used in or as test name tokens        

In [3]:
from prep import * 

In [4]:
text = task_names[0]
text

'Stated Contract Commencement Date (Cl 64)'

In [ ]:
nltk.download('punkt')

In [6]:
import nltk
tokens = nltk.word_tokenize(text)

LookupError: 
**********************************************************************
  Resource punkt not found.
  Please use the NLTK Downloader to obtain the resource:

  >>> import nltk
  >>> nltk.download('punkt')
  
  For more information see: https://www.nltk.org/data.html

  Attempted to load tokenizers/punkt/PY3/english.pickle

  Searched in:
    - '/home/rony/nltk_data'
    - '/home/rony/anaconda3/nltk_data'
    - '/home/rony/anaconda3/share/nltk_data'
    - '/home/rony/anaconda3/lib/nltk_data'
    - '/usr/share/nltk_data'
    - '/usr/local/share/nltk_data'
    - '/usr/lib/nltk_data'
    - '/usr/local/lib/nltk_data'
    - ''
**********************************************************************


### A single hyphen

In [3]:
numbers = [t for t in task_names if len(re.findall(' - ', t))==1]
single_hyphened

['Table 3, Ref 3A - Delivery of the first TBM to HGP',
 'Table 3, Ref 3B - All works in Works Area 1.6',
 'Table 3, Ref 3C - Delivery of the second TBM to HGP',
 'Table 4, Ref 4A - Achieve Deg. 1 for both Up and Down Track Tunnels',
 'Table 1 - Completion of the Whole of the Works (wk13/2015)',
 'Handover of Works Area 1.4B to 826 - first time',
 'Handover of Works Area 1.4B to 826 - second time',
 '830, Trackwork and OHL - Up-track',
 '841A, Signalling System, Trackside Eqpt - Both tunnels',
 '830, Trackwork and OHL - Down-track',
 'A2 (51/2010, 26 Dec 10) - Approval of TBM Tunnelling method Statement',
 'A3 (08/2011, 27 Feb 11) - Implementation of Specified Plans and Programming',
 'A4 (38/2011, 25 Sep 11) - Delivery of 2 nos. TBM to HGP Launching Shaft',
 'A5 (34/2011, 28 Aug 11) - Approval of the MS of Construction of Cross Passages',
 'A5 (34/2011, 28 Aug 11) - Approval of Detailed Interface Require Specific with Designated Contractor',
 'A5 (34/2011, 28 Aug 11) - Implementation o

### Multiple hyphens  

In [4]:
multiple_hyphens = [t for t in task_names if len(re.findall(' - ', t))>1]

In [5]:
multiple_hyphens

['841A - Kwai Chung Vent Bldg (VB6) - SER',
 '841B - Kwai Chung Vent Bldg (VB6) - SER',
 '844 - Kwai Chung Vent Bldg (VB6) - Power supply electrical rooms',
 '830 - Kwai Chung Vent Bldg (VB6) - TSC',
 '849 - Kwai Chung Vent bldg (VB6) - TER, SER, CTER & corridors & roof',
 '851 - Kwai Chung Vent Bldg (VB6) - TER & CTER',
 '855 - Kwai Chung Vent Bldg (VB6) - All rooms & common areas',
 '843 - KC Vent Bldg (VB6) - TVF Rm, compressed & receiver air Rm & Tunnel ECS Control Rm',
 '845 - Kwai Chung Vent Bldg (VB6) - Power supply electrical rooms',
 '850 - Kwai Chung Vent Bldg (VB6) - TER & CTER',
 '853 - Kwai Chung Vent Bldg - TER & CTER',
 'Vault Waterproofing (Ch135030 - 135330) (Bay 33 - 56)',
 'Vault Lining (Ch135614 - 135679, Enlarged section) (Bay 80 - 85)',
 'Vault Lining (Ch135679 - 135712, Enlarged section) (Bay 86 - 87)',
 'Vault Lining (Ch135497 - 135614, Enlarged section) (Bay 69 - 79)',
 'Partition wall (Ch135330.7 - 135712) (Bay 57 - 87)',
 'Partition wall (Ch135060.7 - 135330.

#### Observations     
Hyphens are found to:  
1. Split the name to metadata and activity  
'A10 (34/2013, 25 Aug 13) - Implementation of Specific Plans'   
2. Split the name to an activity and idengtifier   
'Transport TBM components from site to the ferry pier at night - 623N2' 
'855 - Kwai Chung Vehicular Access Adit Rooms' 
3. The activity of interest may be the first or the second part of the name  
4. Indicating what seems as a range of locations, areas   
'Twin track tunnel drill & blast Ch134979 - Ch135279 (300m)'   
5. A task name can contain more than one range
'Tunnel invert & Walkway(UT-Northbound) (Ch136+475~Ch136+656) (Bay F02 - F01)'  

#### Todo:  
<u>Versioning</u>  
*normalized task names*: task names undergoing cleaning and normalization  
<u>Functions</u>:     
1. Identify then filter task names of metadata     
2. Identify then normalize hyphenated ranges    